### ISSUE TITLE
ML: Class imbalance not addressed in LightGBM and CNN training — ~44:1 negative ratio

### ISSUE THREAD

**Issue by roeimed0** [roeimed0/bacterial-gene-prediction#98]
## Problem

The pipeline produces ~176K ORF candidates per genome, of which only ~4K are real genes — a roughly 44:1 negative:positive ratio. Neither the LightGBM \`OrfGroupClassifier\` nor the \`HybridGeneFilter\` CNN+Dense model explicitly handles this imbalance during training:
- No \`scale_pos_weight\` in LightGBM
- No \`pos_weight\` in PyTorch's \`BCEWithLogitsLoss\`
- No over/undersampling

The thresholds (0.1 for LGB, 0.12 for CNN) were lowered from the default 0.5, suggesting the imbalance was noticed post-hoc but never addressed at the training level.

## Impact

Models biased toward predicting "not a gene" (the majority class) will have artificially high accuracy but poor recall. Fixing class imbalance during training typically improves F1 without changing precision.

### LightGBM
\`\`\`python
n_pos = positive_count
n_neg = negative_count
params["scale_pos_weight"] = n_neg / n_pos  # ~44
\`\`\`

### PyTorch (HybridGeneFilter)
\`\`\`python
pos_weight = torch.tensor([n_neg / n_pos])
criterion = nn.BCEWithLogitsLoss(pos_weight=pos_weight)
\`\`\`

After fixing, re-optimise thresholds using a proper precision-recall curve rather than manual tuning.

**Comment by roeimed0** [comment]
Splitting into two focused issues — one per model — to allow independent implementation and testing.
